### PR TITLE
Allow spaces to exist in apple_library::module_map

### PR DESCRIPTION
**Pull request by mattrobmattrob**
Previously, spaces in the value provided to `apple_library::module_map` would crash when running the command for the `extend_modulemap` rule.

```
$ bazel build --apple_platform_type=ios //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap --sandbox_debug
INFO: Analyzed target //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: /Users/mattrobinson/Developer/github/bazel-ios/rules_ios/tests/ios/frameworks/mixed-source/custom-module-map/BUILD.bazel:4:16: error executing shell command: '/bin/bash -c echo "$1" | cat $2 - > $3
module CustomModuleMap.Swift {
    header "CustomModuleMap-Swift.h"
    requires objc
} tests/ios/frameworks/mixed-source/custom-module-map/Module Map/Custo...' failed (Exit 1): sandbox-exec failed: error executing command
  (cd /private/var/tmp/_bazel_mattrobinson/6f4ffdb63927fa472e515bdfa701c08b/sandbox/darwin-sandbox/25/execroot/build_bazel_rules_ios && \
  exec env - \
    TMPDIR=/var/folders/4y/7k0ckw7n69v9bgsvt6x7bnrc0000gn/T/ \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_mattrobinson/6f4ffdb63927fa472e515bdfa701c08b/sandbox/darwin-sandbox/25/sandbox.sb /var/tmp/_bazel_mattrobinson/install/3ac6e4bc67346f686d73708a27e81d33/process-wrapper '--timeout=0' '--kill_delay=15' /bin/bash -c 'echo "$1" | cat $2 - > $3' '' '
module CustomModuleMap.Swift {
    header "CustomModuleMap-Swift.h"
    requires objc
}' 'tests/ios/frameworks/mixed-source/custom-module-map/Module Map/CustomModuleMap.modulemap' bazel-out/ios-x86_64-min13.0-applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap.extended.modulemap) sandbox-exec failed: error executing command
  (cd /private/var/tmp/_bazel_mattrobinson/6f4ffdb63927fa472e515bdfa701c08b/sandbox/darwin-sandbox/25/execroot/build_bazel_rules_ios && \
  exec env - \
    TMPDIR=/var/folders/4y/7k0ckw7n69v9bgsvt6x7bnrc0000gn/T/ \
  /usr/bin/sandbox-exec -f /private/var/tmp/_bazel_mattrobinson/6f4ffdb63927fa472e515bdfa701c08b/sandbox/darwin-sandbox/25/sandbox.sb /var/tmp/_bazel_mattrobinson/install/3ac6e4bc67346f686d73708a27e81d33/process-wrapper '--timeout=0' '--kill_delay=15' /bin/bash -c 'echo "$1" | cat $2 - > $3' '' '
module CustomModuleMap.Swift {
    header "CustomModuleMap-Swift.h"
    requires objc
}' 'tests/ios/frameworks/mixed-source/custom-module-map/Module Map/CustomModuleMap.modulemap' bazel-out/ios-x86_64-min13.0-applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap.extended.modulemap)
cat: tests/ios/frameworks/mixed-source/custom-module-map/Module: No such file or directory
cat: Map/CustomModuleMap.modulemap: No such file or directory
Target //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap failed to build
INFO: Elapsed time: 0.147s, Critical Path: 0.03s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
```

Now it won’t and the extended module map looks as expected:
```
$ bazel build --apple_platform_type=ios //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap
INFO: Analyzed target //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap up-to-date:
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/CustomModuleMap
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Modules/CustomModuleMap.swiftmodule/x86_64.swiftmodule
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/CustomModuleMap-Swift.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/A+CustomModuleMapAdditions.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/A.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/B+CustomModuleMapAdditions.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/B.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/CustomModuleMap.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/PrivateHeaders/PleaseDontImport.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Modules/module.modulemap
INFO: Elapsed time: 0.109s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action

$ cat bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Modules/module.modulemap
framework module CustomModuleMap {
    umbrella header "CustomModuleMap.h"

    module CXX {
        requires cplusplus11
        header "B.h"
    }

    explicit module SuperSecret {
        header "PleaseDontImport.h"
    }

    explicit module A_CustomModuleMapAdditions {
        header "A+CustomModuleMapAdditions.h"
    }

    explicit module B_CustomModuleMapAdditions {
        requires cplusplus11
        header "B+CustomModuleMapAdditions.h"
    }

    export *
    module * { export * }
}

module CustomModuleMap.Swift {
    header "CustomModuleMap-Swift.h"
    requires objc
}
```

### CODE REVIEW

File: rules/library.bzl [hunk 1]
```diff
@@ -68,7 +68,7 @@ module {module_name}.Swift {{
         outputs = [ctx.outputs.destination],
         mnemonic = "ExtendModulemap",
         progress_message = "Extending %s" % ctx.file.source.basename,
-        command = "echo \"$1\" | cat $2 - > $3",
+        command = "echo \"$1\" | cat \"$2\" - > $3",
```
Comment (segiddins) [inline rules/library.bzl:71]: Worth quoting $3 as well?

Comment (segiddins) [inline rules/library.bzl:71]: Worth quoting $3 as well?

**Review by segiddins — APPROVED** [review]

File: rules/library.bzl (hunk 1 above)
Comment (mattrobmattrob) [reply rules/library.bzl]: Seems responsible. Confirmed it still works as expected.

```
$ bazel build --apple_platform_type=ios //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap
Starting local Bazel server and connecting to it...
INFO: Analyzed target //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap (36 packages loaded, 835 targets configured).
INFO: Found 1 target...
Target //tests/ios/frameworks/mixed-source/custom-module-map:CustomModuleMap up-to-date:
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/CustomModuleMap
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Modules/CustomModuleMap.swiftmodule/x86_64.swiftmodule
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/CustomModuleMap-Swift.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/A+CustomModuleMapAdditions.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/A.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/B+CustomModuleMapAdditions.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/B.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Headers/CustomModuleMap.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/PrivateHeaders/PleaseDontImport.h
  bazel-out/applebin_ios-ios_x86_64-dbg-ST-20a50c27621f/bin/tests/ios/frameworks/mixed-source/custom-module-map/CustomModuleMap/CustomModuleMap.framework/Modules/module.modulemap
INFO: Elapsed time: 73.520s, Critical Path: 35.17s
INFO: 245 processes: 19 internal, 224 darwin-sandbox, 1 local, 1 worker.
INFO: Build completed successfully, 245 total actions
```

**Comment by mattrobmattrob** [comment]
Should be good now, @segiddins. Thanks for the fast review.

**Review by justinseanmartin — APPROVED** [review]

**Pull request merged**